### PR TITLE
migration to IP

### PR DESCRIPTION
**Pull request by sanchezl**
KAS-o to etcd currently has a dependency on DNS that is not managed by OpenShift. In order to remove this dependency on an external information, we propose to migrate the KAS-o to etcd to use IP addresses instead of DNS names.

### CODE REVIEW

File: enhancements/etcd/cluster-etcd-operator.md [hunk 1]
```diff
@@ -325,6 +345,42 @@ The cluster can function without intervention, but to fully restore 4.3, manual
     3. wait for the etcd-member to rejoin
     4. move to the next master.
 
+#### Migrating to IPs
+
+Because reasons.
```
Comment (deads2k) [inline enhancements/etcd/cluster-etcd-operator.md]: Cute, but do better.  Avoiding the need to program DNS, removing a dependency on mDNS, avoiding conflicts between the two, providing resilience during upgrades which disrupt the availability of mDNS to respond to DNS name mappings for rebooting etcd endpoints.

File: enhancements/etcd/cluster-etcd-operator.md [hunk 2]
```diff
@@ -325,6 +345,42 @@ The cluster can function without intervention, but to fully restore 4.3, manual
...
+
+##### Etcd Clients
+
+Etcd clients such as KAS-o & OAS-o will compile a list of etcd grpc endpoint urls:
+
+1. From IPs found in `oc -n openshift-etcd get endpoints/host-etcd-2`:
+   * bootstrap member IP from `alpha.installer.openshift.io/etcd-bootstrap` annotation if available.
+   * cluster member IPs from **.spec.subsets[0].addresses[*n*].ip**.
+2. Local host urls: `https://120.0.0.1:2379` and `https://[::1]:2379`
```
Comment (deads2k) [inline enhancements/etcd/cluster-etcd-operator.md]: 127.0.0.1

File: enhancements/etcd/cluster-etcd-operator.md [hunk 3]
```diff
@@ -325,6 +345,42 @@ The cluster can function without intervention, but to fully restore 4.3, manual
...
+
+##### Etcd Serving Certificates
+
+Etcd client serving certificates will need to additionally be valid for `127.0.0.1` and `::1`.
+
+##### Etcd Peer URLs
+
+In release 4.4 etcd **peerUrls** will contain both IP and hostname variations in order to facilitate a downgrade.
```
Comment (deads2k) [inline enhancements/etcd/cluster-etcd-operator.md]: I think we only do this in cases where the DNS name already exists.  Which we should be able to detect.

File: enhancements/etcd/cluster-etcd-operator.md [hunk 4]
```diff
@@ -325,6 +345,42 @@ The cluster can function without intervention, but to fully restore 4.3, manual
...
+
+##### Handling Node IP Changes
+
+If a node's IP address changes, the etcd cluster will become non-functional. We recover by recovering a single etcd
```
Comment (deads2k) [inline enhancements/etcd/cluster-etcd-operator.md]: break this apart into the two cases: 

1. all IPs change at the same time
2. one IP changes.

**Comment by russellb** [comment]
@sanchezl This migration to IPs has already been implemented, right?  Do the contents of this change still accurately reflect what was done?  If so, we should just merge this.  Or is there anyone you'd like to request review from first?

**Pull request closed without merging**